packages/headless/src/composables/tooltip/createTooltip.ts

```diff
@@ -96,16 +96,15 @@ export const createTooltip = createBuilder((options: CreateTooltipOptions) => {
     if (isOutsideClick) _isVisible.value = false;
   };
 
+  document.addEventListener("keydown", handleDocumentKeydown);
+
   /**
    * Registers keydown and click handlers when trigger is "click" to close
    * the tooltip.
    */
   watchEffect(() => {
-    document.removeEventListener("keydown", handleDocumentKeydown);
     document.removeEventListener("click", handleDocumentClick);
-
     if (openType.value === "click") {
-      document.addEventListener("keydown", handleDocumentKeydown);
       document.addEventListener("click", handleDocumentClick);
     }
```

Comment:
We can optimize the code a bit by only removing and adding when necessary.
`addEventListener` won't do anything when the same event and callback combo already exists.
`removeEventListener` won't do anything when the event-listener doesn't exist. 

```vue
  watchEffect(() => {
    if (openType.value === "click") {
      document.addEventListener("click", handleDocumentClick);
    } else {
      document.removeEventListener("click", handleDocumentClick);
    }
```